src/policy/space.rs

```diff
@@ -221,6 +221,19 @@ impl SFTMap {
         }
         self_mut.sft[chunk] = sft;
     }
+
+    pub fn is_in_space(&self, object: ObjectReference) -> bool {
+        let not_in_space = object.to_address().chunk_index() >= self.sft.len()
+            || self.get(object.to_address()).name() == EMPTY_SPACE_SFT.name();
+
+        if not_in_space {
+            // special case - we do not yet have SFT entries for malloc space
+            use crate::policy::mallocspace::is_alloced_by_malloc;
+            is_alloced_by_malloc(object)
```

Comment:
Probably we need a branch here to call `is_alloced_by_malloc` only when MallocMS is the selected plan. Otherwise, I'm not sure if it is safe to inspect malloc related metadata when the object is neither in mmtk heap nor in malloc space.

Reply:
I think it's fine. In `is_alloced_by_malloc()`, there is a check to see if the side metadata is mapped. If not, it won't access the side metadata.